modules/auxiliary/scanner/printer/canon_iradv_pwd_extract.rb

```diff
@@ -0,0 +1,196 @@
+#
+# This module requires Metasploit: http//metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+require 'rex/proto/http'
+require 'msf/core'
+
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Canon IR-Adv Password Extractor',
+      'Description'    => %q{
+        This module will extract the passwords from address books on various Canon IR-Adv mfp devices.
+        Tested models:
+        iR-ADV C2030,
+        iR-ADV 4045,
+        iR-ADV C5030,
+        iR-ADV C5235,
+        iR-ADV C5240,
+        iR-ADV 6055,
+        iR-ADV C7065
+      },
+      'Author'         =>
+        [
+          'Deral "Percentx" Heiland',
+          'Pete "Bokojan" Arzamendi'
+        ],
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        OptBool.new('SSL', [true, "Negotiate SSL for outgoing connections", false]),
+        OptInt.new('ADDRSBOOK', [ true, 'The number of the address book to extract 1-11', 1]),
+        OptInt.new('RPORT', [ true, 'The target port', 8000]),
+        OptString.new('USER', [ true, 'The default Admin user', '7654321']),
+        OptString.new('PASSWD', [ true, 'The default Admin password', '7654321']),
+        OptInt.new('TIMEOUT', [true, 'Timeout for printer probe', 20])
+
+      ], self.class)
+  end
+
+# Time to start the fun
```

Comment:
Run a module is fun, IMHO, I think that do not need to describe this in code. :P